File: Array/p274_H_Index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort(reverse=True)
        print(citations)

        h=0
        for i, c in enumerate(citations):
            if c >= i+1:
                h= i+1
        
        return h

    #time complexity O(N)
    #space complexity O(1)

        

File: Array/p274_H_Index.cpp
class Solution {
public:
  int hIndex(vector<int> &citations) {
    int res = 0;
    sort(citations.begin(), citations.end());
    for (int i = 0; i < citations.size(); i++) {
      int remain = citations.size() - (i);
      res = max(res, min(citations[i], remain));
    }
    return res;
  }
};
// time complexity O(N logN)
// space complexity O(1)